tests/quality_pragmaitc/test_task.py

```diff
@@ -16,3 +17,49 @@ def test_shouldProvideCorrectValueForMetric():
     task.setProvidedQuality(baseline, CommonMetrics.METERS, 50.0)
 
     assert 50.0 == task.myProvidedQuality(CommonMetrics.METERS, fullContext)
+
+
+def test_metricNotFound():
+    task = Task()
+    currentContext = Context("C1")
+    fullContext = set()
+
+    fullContext.add(currentContext)
+
+    task.setProvidedQuality(currentContext, CommonMetrics.METERS, 30.0)
+
+    try:
+        task.myProvidedQuality(CommonMetrics.SECONDS, fullContext)
+        assert False
+
+    except MetricNotFoundException as exp:
+        assert exp.message == 'Metric: SECONDS'
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=4895731)